examples/tutorial_server_eventfilter.c

```diff
@@ -0,0 +1,282 @@
+static UA_StatusCode
+generateEventsMethodCallback(UA_Server *server,
+                         const UA_NodeId *sessionId, void *sessionHandle,
+                         const UA_NodeId *methodId, void *methodContext,
+                         const UA_NodeId *objectId, void *objectContext,
+                         size_t inputSize, const UA_Variant *input,
+                         size_t outputSize, UA_Variant *output) {
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Creating event");
+    UA_StatusCode retval;
+    UA_NodeId* eventNodeId;
+    eventNodeId = (UA_NodeId*)
+        UA_Array_new(eventTypesCount, &UA_TYPES[UA_TYPES_NODEID]);
+    /* set up event */
+    for(size_t i = 0 ; i < eventTypesCount; i++) {
+        retval = setUpEvent(server, &eventNodeId[i], eventTypes[i], false,i);
+        if(retval != UA_STATUSCODE_GOOD) {
+            UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                           "Creating event failed. StatusCode %s", UA_StatusCode_name(retval));
+            return retval;
+        }
+    }
+    for(size_t i = 0 ; i < eventTypesCount; i++) {
+        retval = UA_Server_triggerEvent(server, eventNodeId[i],
+                                         UA_NODEID_NUMERIC(0, UA_NS0ID_SERVER),
+                                         NULL, UA_TRUE);
+        if(retval != UA_STATUSCODE_GOOD)
+            UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                           "Triggering event failed. StatusCode %s", UA_StatusCode_name(retval));
+    }
+    return retval;
+}
+
+static UA_StatusCode
+generateRandomEventMethodCallback(UA_Server *server,
+                             const UA_NodeId *sessionId, void *sessionHandle,
+                             const UA_NodeId *methodId, void *methodContext,
+                             const UA_NodeId *objectId, void *objectContext,
+                             size_t inputSize, const UA_Variant *input,
+                             size_t outputSize, UA_Variant *output) {
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Creating event");
+    size_t random = (size_t) rand() % eventTypesCount;
+    /* set up event */
+    UA_NodeId eventNodeId;
+    UA_StatusCode retval = setUpEvent(server, &eventNodeId, eventTypes[random], true,random);
+    if(retval != UA_STATUSCODE_GOOD) {
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                       "Creating event failed. StatusCode %s", UA_StatusCode_name(retval));
+        return retval;
+    }
+    retval = UA_Server_triggerEvent(server, eventNodeId,
+                                    UA_NODEID_NUMERIC(0, UA_NS0ID_SERVER),
+                                    NULL, UA_TRUE);
+    if(retval != UA_STATUSCODE_GOOD)
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                       "Triggering event failed. StatusCode %s", UA_StatusCode_name(retval));
+    return retval;
+}
```

Comment:
You're defining a lot of complex events but you don't write the properties here. Is there any reason? Because we could youse more interesting where clauses if we have properties with different values.

Reply:
The properties are simply set in another method (see setUpEvent) for the sake of clarity. 
There is an option to randomize the values of the properties by passing true to the setUpEvent method in the boolean argument "random".